compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1298,11 +1298,16 @@ TR::Register *OMR::X86::TreeEvaluator::iternaryEvaluator(TR::Node *node, TR::Cod
    TR::Register *falseReg = cg->evaluate(falseVal);
    bool trueValIs64Bit = TR::TreeEvaluator::getNodeIs64Bit(trueVal, cg);
    TR::Register *trueReg  = TR::TreeEvaluator::intOrLongClobberEvaluate(trueVal, trueValIs64Bit, cg);
+   if (!node->isNotCollected()) 
+      trueReg->setContainsCollectedReference();
 
    // don't need to test if we're already using a compare eq or compare ne
    auto conditionOp = condition->getOpCode();
+   bool longCompareOn32bit = (conditionOp.isBooleanCompare() &&
+         (condition->getFirstChild()->getOpCode().isLong() || condition->getSecondChild()->getOpCode().isLong())
```

Comment:
The binary IL opcode of condition should always expect both its children have same type; so I guess checking both children of condition is redundant.

Reply:
addressed.